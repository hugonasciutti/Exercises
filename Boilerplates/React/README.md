# Features

Dry boiler plate with:

- Webpack dev server
- React
- Babel Env
