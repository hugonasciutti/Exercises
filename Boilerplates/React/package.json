{
  "name": "boilerplatereact",
  "version": "1.0.0",
  "description": "",
  "main": "index.js",
  "scripts": {
    "start": "webpack-dev-server"
  },
  "author": "",
  "license": "ISC",
  "dependencies": {
    "final-form": "^4.5.2",
    "react": "^16.3.2",
    "react-dom": "^16.3.2"
  },
  "devDependencies": {
    "babel-core": "^6.26.0",
    "babel-loader": "^7.1.4",
    "babel-preset-env": "^1.6.1",
    "babel-preset-react": "^6.24.1",
    "babel-preset-stage-0": "^6.24.1",
    "html-webpack-plugin": "^3.2.0",
    "webpack": "^4.6.0",
    "webpack-cli": "^2.0.15",
    "webpack-dev-server": "^3.1.3"
  }
}
