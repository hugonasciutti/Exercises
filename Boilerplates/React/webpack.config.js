const HtmlWebpackPlugin = require('html-webpack-plugin')

module.exports = {
  mode: 'development',

  entry: './src/index.jsx',

  output: {
    path: `${__dirname}/dist`,
    filename: 'app.js'
  },

  devtool: 'source-map',

  module: {
    rules: [
      { test: /\.(js|jsx)?$/, loader: 'babel-loader', exclude: /node_modules/ }
    ]
  },
  plugins: [
    new HtmlWebpackPlugin({
      title: 'Boilerplate',
      template: './src/index.ejs'
    })
  ]
}
