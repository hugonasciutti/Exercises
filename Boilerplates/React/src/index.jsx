import React, { Component } from 'react'
import { render } from 'react-dom'
import Form from './Form'

class App extends Component {
  render() {
    return (
      <div>
        <Input />
      </div>
    )
  }
}

render(<App />, document.getElementById('app'))