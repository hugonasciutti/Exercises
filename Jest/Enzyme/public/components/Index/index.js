import React, { Component } from 'react';

class Index extends Component {
  deleteItem(item, position) {
    return item.split(position, 1);
  }
  render() {
    const items = ['one', 'two', 'three'];
    return (
      <div>
        <h1>This is the Index Page</h1>
        {items.map((item, index) => <h2 key={`item-${index + 1}-index`}>{item}</h2>)}
      </div>
    );
  }
}

export default Index;
