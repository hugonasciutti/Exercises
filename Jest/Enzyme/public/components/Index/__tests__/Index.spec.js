import React from 'react';
import { shallow } from 'enzyme';
import Index from '../index';

describe('<Index />', () => {
  it('should render <Index /> component', () => {
    const wrapper = shallow(<Index />);
    expect(wrapper).toMatchSnapshot();
  });

  it('Immutable Page\'s Title', () => {
    const wrapper = shallow(<Index />);
    const title = wrapper.find('#title').node.props.children;
    expect(title).toBe('Index Title');
  });
});
