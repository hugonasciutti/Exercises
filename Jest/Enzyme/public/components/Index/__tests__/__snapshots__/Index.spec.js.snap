// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Index /> should render <Index /> component 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div>
    <h1
        id="title"
    >
        Index Title
    </h1>
    <h2>
        one
    </h2>
    <h2>
        two
    </h2>
    <h2>
        three
    </h2>
</div>,
  "nodes": Array [
    <div>
      <h1
            id="title"
      >
            Index Title
      </h1>
      <h2>
            one
      </h2>
      <h2>
            two
      </h2>
      <h2>
            three
      </h2>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Index />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Index {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {},
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div>
          <h1
                    id="title"
          >
                    Index Title
          </h1>
          <h2>
                    one
          </h2>
          <h2>
                    two
          </h2>
          <h2>
                    three
          </h2>
</div>,
        "_debugID": 2,
        "_renderedOutput": <div>
          <h1
                    id="title"
          >
                    Index Title
          </h1>
          <h2>
                    one
          </h2>
          <h2>
                    two
          </h2>
          <h2>
                    three
          </h2>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Index />,
}
`;
